### DISCUSSION TITLE
Error in allocating memory parameter (mem_limit) when using Docker Compose to start the service

### DISCUSSION THREAD

**Discussion by ls17680477939** [moby/moby#45807] [Q&A]
Partial version environment：
Docker version 19.03.8
docker-compose version 1.29.2
Linux-kernel 6.3.9-1.el7.elrepo.x86_64

Execute Command：docker-compose up -d

ERROR: for iship-ams  Cannot start service iship-ams: OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:297: applying cgroup configuration for process caused \"failed to write 1 to memory.kmem.limit_in_bytes: write /sys/fs/cgroup/memory/docker/9c52264a200e14f2a137ba3a77e902aa6f8ffb41d493ea9d30c1e43822d4172a/memory.kmem.limit_in_bytes: operation not supported\"": unknown
ERROR: Encountered errors while bringing up the project.

**Comment by thaJeztah** [comment]
It looks like you're running a very old (and EOL) version of the Docker Engine. Kernel memory limits have been deprecated in the Linux kernel, and deprecated in the Docker Engine (and runc); https://docs.docker.com/engine/deprecated/#kernel-memory-limit

Older versions of the Docker Engine may be (implicitly) setting that option, or perhaps your compose file has the option set. Are you still seeing this option on a currently maintained version of the Docker Engine (and containerd, runc)?
